low-level/sensors/impl/generic/raid.py

```diff
@@ -224,7 +236,8 @@ def _notify_NodeDataMsgHandler(self):
                             resource_id = device+":/dev/"+drive_name
                             drive_status = drive.get("status")
                             if drive_status not in ["U", "UP"] and device in self._faulty_drive_list and \
-                                drive_name not in self._faulty_drive_list[device]:
+                                drive_name not in self._faulty_drive_list[device] and \
+                                self.prev_alert_type[device] != self.MISSING:
```

Comment:
Am confused here. You are checking whether previous alert for whole raid array device was not missing?